prismarine-viewer/viewer/lib/entities.ts

```diff
@@ -467,9 +492,17 @@ export class Entities extends EventEmitter {
     }
     // ---
     // not player
-    const displayTextRaw = meta.custom_name_visible || getSpecificEntityMetadata('text_display', entity)?.text
-    const displayText = displayTextRaw && this.parseEntityLabel(displayTextRaw)
+    const textDisplayMeta = getSpecificEntityMetadata('text_display', entity)
+    const displayTextRaw = textDisplayMeta?.text || meta.custom_name_visible && meta.custom_name
+    const displayText = this.parseEntityLabel(displayTextRaw)
     if (entity.name !== 'player' && displayText) {
+      if (textDisplayMeta) {
+        this.entitiesOptions.nameTagBackgroundColor = toRgba(textDisplayMeta.background_color)
+        if (textDisplayMeta.text_opacity) {
+          const rawOpacity = parseInt(textDisplayMeta.text_opacity, 10)
+          this.entitiesOptions.nameTagTextOpacity = rawOpacity > 0 ? rawOpacity : 256 - rawOpacity
```

Comment:
thats a wrong idea to expand the usage of `entitiesOptions` here. Why would we ever store it in a global scope of the class? Since it's bound to a specific entity we can simply use a variable locally and pass it later to the render function (addNametag). I think we just don't add props to entitiesOptions - these are meant for global options

As I understand right now, nametag of some other entity might use the style of the last rendered text_display:
https://github.com/zardoy/prismarine-web-client/blob/29833019154b7b201867c11ed59158ba625b1cf0/prismarine-viewer/viewer/lib/entities.ts#L451

Reply:
For some reason I thought this was instantiated for each individual entity. Will adjust it.